File: Cases/Case3/Go/main.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func main() {
	rand.Seed(time.Now().Unix())
	var testArray [200000]int
	//Fills testArray with random numbers between 0 and 999999
	for index := 0; index < len(testArray); index++ {
		testArray[index] = rand.Intn(1000000)
	}
	//SearchedNum is assigned the last value in the array to create the worst case
	var searchedNum = testArray[len(testArray)-1]
	var foundFlag = false

	//Timer start
	start := time.Now()
	//Sequential search for searchedNum
	for index := 0; index < len(testArray); index++ {
		if searchedNum == testArray[index] {
			foundFlag = true
		}
	}

	fmt.Println(foundFlag)
	fmt.Println(time.Since(start))
}

File: Cases/Case3/Rust/src/main.rs
extern crate rand;

use rand::Rng;
use std::time::Instant;

fn main() {
    
    let mut testArray=[0;200000];
    
    //Fills testArray with random numbers between 0 and 999999
    for index in 0..testArray.len(){
        testArray[index]=rand::thread_rng().gen_range(0,1000000);
    }
    
	//SearchedNum is assigned the last value in the array to create the worst case
    let searchedNum=testArray[testArray.len()-1]; 
    let mut foundFlag = false;
    //Timer start
    let start = Instant::now();
    
	//Sequential search for searchedNum
    for index in 0..testArray.len(){
        if searchedNum==testArray[index]{
            foundFlag=true;
        }
    }
    println!("{}", foundFlag);
    println!("{}", start.elapsed().as_millis());
}

